lib/em/pure_ruby.rb

```diff
@@ -33,6 +33,18 @@
 require 'socket'
 require 'fcntl'
 require 'set'
+require 'openssl'
+
+# Older versions of Ruby & OpenSSL may not provide the
+# SSLErrorWaitReadable and SSLErrorWaitWritable OpenSSL classes.
+# Set them if not defined.
+unless defined?(OpenSSL::SSL::SSLErrorWaitReadable)
+  OpenSSL::SSL::SSLErrorWaitReadable = OpenSSL::SSL::SSLError
```

Comment:
The problem is that this will rescue instances of OpenSSL::SSL::SSLError in the blocks for the Read/Write errors. I think the way to roll is to flip the situation around, so below you'd rescue a new class `EventMachine::Error::SSLErrorWaitWritable`, and then define it as:

``` ruby
if defined?(OpenSSL::SSL::SSLErrorWaitWritable)
  EventMachine::Error::SSLErrorWaitWritable = OpenSSL::SSL::SSLErrorWaitWritable
else
  class EventMachine::Error::SSLErrorWaitWritable < EventMachine::Error; end
end
```